SRC/bank_01f.asm

```diff
@@ -0,0 +1,578 @@
+					;if validate required, check if item has been collected (=$02)
+					;if not collected, is validated so draw
+						ld a, [bc]
+						cp a, set_item_collected
+						jr z, .isCollected
+						jr .itemValidated
+						;if item SRAM isn't set to COLLECTED, check working RAM
+;							ld b, $c5
+;							ld a, c
+;							and a, $3F
+;							add a, $40
+;							ld c, a
+;							ld a, [bc]
+;							cp a, COLLECTED
+;							jr nz, .itemValidated
+						;if collected, remove this sprite and loop
+						.isCollected:
+							inc hl
+							inc hl
+							inc hl
+							inc hl
+							ld a, tile_blank
+							ld [de], a
+							inc de
+							ld [de], a
+							inc de
+							ld [de], a
+							inc de
+							ld [de], a
+							inc de
+							jr .loopSetupMapSprites
+				;at this point we are loading sprites	
+				.itemValidated:
+					ld a, [hl+]
+					ld [de], a
+					inc de
+					ld a, [hl+]
+					ld [de], a
+					inc de
+					ld a, [hl+]
+					ld [de], a
+					inc de
+					ld a, [hl+]
+					ld [de], a
+					inc de
+				jr .loopSetupMapSprites
+			.setupMapSpritesExit:
+				
+	;set window
+	ld a, windowHeight_map
+	ldh [rWY], a
+ret
+
+;sprite setup
+;$C000..9F: OAM
+;    + 0: Y position
+;    + 1: X position
+;    + 2: Tile number
+;    + 3: Attributes
+;         10: Palette
+;         20: X flip
+;         40: Y flip
+;         80: Priority (set: behind background)
+
+
+;loading items:
+;c900-cabf is $40 bytes per bank (in debugger 4 lines), 7 banks requires through CABF
+;offset is 40 * (bank-9) + c900
+;if we know each value of each item, we can have an ID value associated with it that gets loaded
+;id is the index to add to our entry point of c900, c940, c980, c9c0, ca00, ca40, ca80
+;if that indexed value does not equal FF, do not draw the dot
+
+
+;below - here, we handle the items-found counter,
+;then we load the text for the appropriate item if it's not a refill
+m2maps_calcFoundEquipment:
+		ld a, b
+		cp isRefill
+		jr nc, .isRefill
+		cp isBeam
+		jr nc, .gotItem
+			ld a, [samusBeam]
+			cp isFirstBeam
+			jr nz, .notFirstBeam
+			jr .doNotClearItem
+		.gotItem:
+			;get sram bank offset from enemy wram
+				ld a, [clearItemDotLow]
+				add a, clearItem_enemyOffset
+				ld e, a
+				ld a, [clearItemDotHigh]
+				ld d, a
+				ld a, [de]
+				sub a, clearItem_sramOffsetLo
+				ld [clearItemIndex], a
+			;set up bank as high byte of de for SRAM address, is $bank - $9 + $c9
+				ld a, [currentLevelBank]
+				sub a, clearItem_bankOffset
+				sra a
+				sra a
+				add a, clearItem_sramOffsetHi
+				ld d, a
+				ld [clearItemBank], a
+			;loop to set up low byte of de with offset for enemy index for SRAM low address
+				ld a, [currentLevelBank]
+				sub a, clearItem_bankOffset
+				inc a
+				and a, clearItem_loopLimit
+				ld e, a
+				;get a for loop once more
+				ld a, [clearItemIndex]
+				.loopCurrentCheck
+					dec e
+					jr z, .stopChecking
+					add a, clearItem_sramOffsetLo
+					jr .loopCurrentCheck
+				.stopChecking:
```

Comment:
If `[currentLevelBank]` is Ch, then the loop runs a good 255 times (and then happens to churn out the correct result of `a = C0h`), looks like the `inc a` is supposed to follow the `and`.